pandas/core/common.py

```diff
@@ -1613,20 +1614,26 @@ def interpolate_1d(xvalues, yvalues, method='linear', limit=None,
                              'DatetimeIndex')
         method = 'values'
 
-    def _interp_limit(invalid, limit):
+    def _interp_limit(invalid, fw_limit, bw_limit):
         """mask off values that won't be filled since they exceed the limit"""
         all_nans = np.where(invalid)[0]
         if all_nans.size == 0: # no nans anyway
             return []
-        violate = [invalid[x:x + limit + 1] for x in all_nans]
-        violate = np.array([x.all() & (x.size > limit) for x in violate])
-        return all_nans[violate] + limit
+        violate = [invalid[x - bw_limit:x + fw_limit + 1] for x in all_nans]
+        violate = np.array([x.all() & (x.size > bw_limit + fw_limit)
+                            for x in violate])
+        return all_nans[violate] + fw_limit - bw_limit
 
     xvalues = getattr(xvalues, 'values', xvalues)
     yvalues = getattr(yvalues, 'values', yvalues)
 
     if limit:
-        violate_limit = _interp_limit(invalid, limit)
+        fw_limit = bw_limit = limit
+        if limit_direction == 'forward':
```

Comment:
Both would be good ideas!

On Fri, Aug 14, 2015 at 7:36 PM, Leif Johnson <notifications@github.com>
wrote:

>>  
>>      xvalues = getattr(xvalues, 'values', xvalues)
>>      yvalues = getattr(yvalues, 'values', yvalues)
>>  
>>      if limit:
>> -        violate_limit = _interp_limit(invalid, limit)
>> +        fw_limit = bw_limit = limit
>> +        if limit_direction == 'forward':
> Do you mean I should catch the 'both' case explicitly? Or convert limit_direction to lowercase before comparing to 'forward'/'backward'?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/pydata/pandas/pull/10691/files#r37132141